rust-wrapper/src/msm.rs

```diff
@@ -0,0 +1,76 @@
+use ark_bls12_381::{Fr as ScalarField, G1Affine as GAffine, G1Projective as G};
+use ark_ec::VariableBaseMSM as BaselineVariableBaseMSM;
+use ark_ff::PrimeField;
+use ark_msm::msm::VariableBaseMSM;
+use ark_serialize::CanonicalSerialize;
+use ark_std::log2;
+use haskell_ffi::{from_haskell::marshall_from_haskell_var, to_haskell::marshall_to_haskell_var};
+use utils::{Buffer, RW};
+
+use crate::utils::{self, deserialize_vector_points, deserialize_vector_scalar_field};
+
+#[no_mangle]
+pub extern "C" fn rust_wrapper_multi_scalar_multiplication(
+    points_var: *const u8,
+    points_len: usize,
+    scalars_var: *const u8,
+    scalars_len: usize,
+    out: *mut u8,
+    out_len: &mut usize,
+) {
+    let scalars: Vec<ScalarField> =
```

Comment:
`rustc` should be able to infer these type annotations, can it not?

Reply:
fixed